src/Engine/ProtoCore/FFI/CLRDLLModule.cs

```diff
@@ -393,18 +403,32 @@ private ClassDeclNode ParseSystemType(Type type, string alias)
             
             foreach (var m in methods)
             {
+                var na = m.Name;
+
                 if (SupressesImport(m, mGetterAttributes))
                     continue;
 
-                if (classnode.IsStatic && m.GetBaseDefinition().DeclaringType == baseType && baseType == typeof(object))
+                MethodInfo baseDefinition;
+                if (m.DeclaringType.Assembly.ReflectionOnly)
+                {
+                    baseDefinition = m.GetBaseDefinitionReflectionContext();
+                    if (baseDefinition is null)
+                        continue;
```

Comment:
yea so the main if statement here uses `GetBaseDefinitionReflectionContext` which can potentially return null as there are cases where some of the checks that it performs fails because of the reflection context. In the `else` statement we are using `GetBaseDefinition` which will never return null, it will either return the base definition or the same methodInfo. Which is why im only checking for null here.